src/Features/Core/Portable/Diagnostics/WorkspaceAnalyzerOptions.cs

```diff
@@ -10,13 +14,35 @@ namespace Microsoft.CodeAnalysis.Diagnostics
     /// </summary>
     internal sealed class WorkspaceAnalyzerOptions : AnalyzerOptions
     {
-        public WorkspaceAnalyzerOptions(AnalyzerOptions options, Workspace workspace)
+        private readonly Workspace _workspace;
+        private readonly OptionSet _optionSet;
+
+        public WorkspaceAnalyzerOptions(AnalyzerOptions options, OptionSet optionSet, Workspace workspace)
             : base(options.AdditionalFiles)
         {
-            this.Workspace = workspace;
+            _workspace = workspace;
+            _optionSet = optionSet;
         }
 
-        public Workspace Workspace { get; }
+        public HostWorkspaceServices Services => _workspace.Services;
+
+        public async Task<OptionSet> GetDocumentOptionSetAsync(SyntaxTree syntaxTree, CancellationToken cancellationToken)
+        {
+            var documentId = _workspace.CurrentSolution.GetDocumentId(syntaxTree);
+            if (documentId == null)
+            {
+                return _optionSet;
+            }
+
+            var document = _workspace.CurrentSolution.GetDocument(documentId);
+            if (document == null)
+            {
+                return _optionSet;
+            }
+
+            var documentOptionSet = await document.GetOptionsAsync(_optionSet, cancellationToken).ConfigureAwait(false);
+            return documentOptionSet ?? _optionSet;
```

Comment:
sure. but the original code was with ??
..
var documentOptionSet = await document.GetOptionsAsync(cancellationToken).ConfigureAwait(false);		 +            return workspaceAnalyzerOptions.GetDocumentOptionSetAsync(syntaxTree, cancellationToken);
 -            return documentOptionSet ?? workspace.Options;
..